src/reduce.test.js

```diff
@@ -11,9 +11,39 @@ describe('reduce', () => {
     delete Array.prototype.reduce2;
   });
 
-  it('should ', () => {
+  it('should calculate the sum of array elements', () => {
+    const arr = [1, 2, 3, 4, 5];
+    const sum = arr.reduce2((acc, curr) => acc + curr, 0);
 
+    expect(sum).toBe(15);
   });
 
-  // Add tests here
+  it('should combine array elements into a string', () => {
+    const arr = ['Hello', ' ', 'world', '!'];
+    const str = arr.reduce2((acc, curr) => acc + curr, '');
+
+    expect(str).toBe('Hello world!');
+  });
+
+  it('should find max element of the array', () => {
+    const arr = [10, 5, 8, 3, 12];
+    const max = arr.reduce2((acc, curr) => (curr > acc ? curr : acc));
+
+    expect(max).toBe(12);
+  });
+
+  it('should count average value of the array', () => {
+    const arr = [2, 4, 6, 8, 10];
+    const avg = arr.reduce2((acc, curr, index, array) => {
+      const total = acc + curr;
+
+      if (index === array.length - 1) {
```

Comment:
Avoid using `else` after a `return` statement. You can directly use `return total;` outside the `if` block.